.github/workflows/test.yml

```diff
@@ -33,3 +33,23 @@ jobs:
       run: |
         pytest
 
+  docs:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout source
+      uses: actions/checkout@v2
+
+    - name: Test
+      run: |
+        echo "::add-matcher::./.github/workflows/problem-matchers/doxygen.json"
+        # Problem matches don't actually error a step, so we have to manually
+        # check for warnings/errors at the end. Capture the output so we can
+        # parse it later.
+        make -C doc 2>&1 | tee doxygen-log.txt
+        echo "::remove-matcher owner=doxygen::"
```

Comment:
[question] Is removing the matcher necessary?  

I checked a couple of the repos under [Examples in the docs](https://github.com/actions/toolkit/blob/main/docs/problem-matchers.md#examples) and they don't remove their matcher.

One example I found of `remove-matcher` was to disable the built-in eslint matcher set up by `setup-npm` in https://github.com/actions/setup-node/issues/72

Is it perhaps that matchers live on beyond the step/action?

Reply:
Added a note [here](https://github.com/TheFoundryVisionmongers/OpenAssetIO/pull/104/commits/6764e7a1b770840bfe3c2d281170e56c9d30dcd4) on this.